

to create environment with python verstion
```
$ conda create -p venv python=3.8 -y
```


to activate evnironment venv
```
$ source activate ./venv
```


to initialize git
```
$ git init
```
goto source control in vs code and then commit, it will publish all files of ur branch in git



touch will create the files
```
$ touch .gitignore
$ touch .README.md
```

create requirements.txt file
```
$ touch requirements.txt
to install  libraries at requirements.txt file
$ pip install -r requirements.txt
```

create stage_01,02.py files then initialize dvc
```
$ dvc init
it will create .dvcignore file auto
```

artifact.txt will create if u run stage_01.py
```
$ python stage_01.py
print the text at artifact.txt file after run stage_02.py
$ python stage_02.py
```

create dvc.yaml file at dvc folder
run dvc repro by removing artifact.txt
```
$ dvc repro
it will create artifact.txt ,dvc.lock and it shows stage_01,02.py files values
```
```
$ dvc dag
it shows dependencies vth stage_01 to 02.py file
stage_02 depends on stage_ 01.py
stage_03 depends on stage_ 01.py

```
create test.ipynb file by create folder notebook
create dataframe, save it as csv file in data folder
add the data folder by dvc
```
$ dvc add notebook/add
it will create data.dvc file
```
commit the data.dvc folder to git
```
$ git add notebook/data.dvc && git commit -m "add data file succesfully"
```

create tmp/dvcstore folder
```
$ pwd
$ mkdir tmp
$ cd tmp
$ pwd
$ mkdir dvcstore
$pwd
it will create tmp/dvcstore folder
```
create the remote storage space vth myremote
```
$ dvc remote add -d myremote /tmp/dvcstore
```
push the dvc
here 3 tmp files created,stored in wrong path,needs to be cleared, its remote path problem
```
$ dvc push
```